woocommerce/woocommerce-gutenberg-products-block
# Build: Add translation conversion script

Following on from #339; this PR adds a script to generate these JSON files from [the language pack .po files.](https://translate.wordpress.org/projects/wp-plugins/woo-gutenberg-products-block/language-packs) The relevant bit from that PR:

> Unfortunately, it seems that wp.org isn't generating the JSON files needed for the translations. Until that's the case, we can bundle the complete translations in the plugin. [...] So we've added 8 copies of the same JSON, one for each block JS file.

By default, the script will look for `.po` files in the `languages` directory, though we don't check these in. Instead, I was using `bin/languages` to test with. You can also optionally pass `--debug` to see more verbose output.

### How to test the changes in this Pull Request:

1. Download [some language packs](https://translate.wordpress.org/projects/wp-plugins/woo-gutenberg-products-block/language-packs) into `bin/languages`
2. Run `./bin/generate-translation-json.js --source ./bin/languages/ --debug`
3. Expect: you should see a few "Converting…" and "Writing…" messages
4. There should be no change to the translation files (since all _should_ be up to date).
